fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -392,22 +396,50 @@ public void updateUniquenessViolations(@Nonnull Tuple valueKey, @Nonnull Tuple p
      */
     @Nonnull
     protected RecordCursor<InvalidIndexEntry> validateOrphanEntries(@Nullable byte[] continuation,
-                                                          @Nullable ScanProperties scanProperties) {
-        if (scanProperties == null) {
-            scanProperties = new ScanProperties(ExecuteProperties.newBuilder()
-                    .setReturnedRowLimit(Integer.MAX_VALUE)
-                    // For orphan index entry validation, it does not hurt to have a weaker isolation.
-                    .setIsolationLevel(IsolationLevel.SNAPSHOT)
-                    .build());
-        }
+                                                                    @Nonnull ScanProperties scanProperties) {
         return scan(IndexScanType.BY_VALUE, TupleRange.ALL, continuation, scanProperties)
                 .filterAsync(
                         indexEntry -> state.store
                                 .hasIndexEntryRecord(indexEntry, IsolationLevel.SNAPSHOT)
                                 .thenApply(has -> !has),
-                        FDBRecordStore.DEFAULT_PIPELINE_SIZE)
-                .map(indexEntry ->
-                        new InvalidIndexEntry(indexEntry, InvalidIndexEntry.Reasons.ORPHAN));
+                        DEFAULT_PIPELINE_SIZE)
+                .map(InvalidIndexEntry::newOrphan);
+    }
+
+    /**
+     * Validate entries in the index. It scans the records and checks if the index entries associated with each record
+     * exist. Note that it may not work for synthetic indexes.
```

Comment:
You should clarify what you mean by "synthetic indexes".  I think you probably mean indexes that act as materialized views and do not actually point to base records, such as aggregate indexes.